lib/streamlit/__init__.py

```diff
@@ -480,6 +482,58 @@ def experimental_show(*args):
         exception(exc, exc_tb)  # noqa: F821
 
 
+def experimental_get_query_params():
+    """Return the query parameters that is currently showing in the browser's URL bar.
+
+    Returns
+    -------
+    dict
+      The current query parameters as a dict. "Query parameters" are the part of the URL that comes
+      after the first "?".
+
+    Example
+    -------
+
+    Let's say the user's web browser is at
+    `http://localhost:8501/?show_map=true&number_of_countries=2`. Then, you can get the query
+    parameters using the following:
+
+    >>> st.experimental_get_query_params()
+    {"show_map": ["true"], "number_of_countries": ["2"]}
+
+    """
+    ctx = _get_report_ctx()
+    if ctx is None:
+        return ""
+    return _parse.parse_qs(ctx.query_string)
+
+
+def experimental_set_query_params(query_params):
+    """Set the query parameters that are shown in the browser's URL bar.
+
+    Parameters
+    ----------
+    query_params : dict
+        The query parameters to set, as key-value pairs.
+
+    Example
+    -------
+
+    To point the user's web browser to something like
+    `http://localhost:8501/?show_map=true&number_of_countries=2`, you would do the following:
+
+    >>> st.experimental_set_query_params({"show_map": "true", "number_of_countries": 2})
```

Comment:
Do we handle passing lists as values? Should we include that scenario in the examples?

Reply:
Good point. Done.